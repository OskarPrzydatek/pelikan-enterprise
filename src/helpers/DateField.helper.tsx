import React from 'react';
import { FormProvider, useForm } from 'react-hook-form';

import { DateField } from '~/components/molecules';
import { StyledComponentsProvider } from '~/providers';

interface IDateFieldHelper {
  isError?: boolean;
}

export const DateFieldHelper: React.FC<IDateFieldHelper> = ({
  isError,
}: IDateFieldHelper) => {
  const methods = useForm<{ startDate: string }>();

  React.useEffect(() => {
    if (isError)
      methods.setError('startDate', {
        type: 'required',
        message: 'This is required',
      });
  }, [isError, methods]);

  return (
    <StyledComponentsProvider>
      <FormProvider {...methods}>
        <DateField
          errorMessageTestID="molecule-datefield-error-message"
          inputTestID="molecule-datefield"
          label="Od"
          labelTestID="molecule-datefield-label"
          name="startDate"
          registerOptions={{}}
        />
      </FormProvider>
    </StyledComponentsProvider>
  );
};
