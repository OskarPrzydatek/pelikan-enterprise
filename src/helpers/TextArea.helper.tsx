import React from 'react';
import { FormProvider, useForm } from 'react-hook-form';

import { StyledComponentsProvider } from '~/styles';

import { TextArea } from '../components/molecules/TextArea/TextArea.component';

interface ITextAreaHelper {
  isError?: boolean;
}

export const TextAreaHelper: React.FC<ITextAreaHelper> = ({
  isError,
}: ITextAreaHelper) => {
  const methods = useForm<{ description: string }>();

  React.useEffect(() => {
    if (isError)
      methods.setError('description', {
        type: 'required',
        message: 'This is required',
      });
  }, [isError, methods]);

  return (
    <StyledComponentsProvider>
      <FormProvider {...methods}>
        <TextArea
          errorMessageTestID="molecule-textarea-error-message"
          label="Description"
          labelTestID="molecule-textarea-label"
          name="description"
          registerOptions={{}}
          textareaTestID="molecule-textarea"
        />
      </FormProvider>
    </StyledComponentsProvider>
  );
};
