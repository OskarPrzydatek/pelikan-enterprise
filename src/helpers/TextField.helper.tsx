import React from 'react';
import { useForm } from 'react-hook-form';

import { StyledComponentsProvider } from '~/styles';

import { TextField } from '../components/molecules/TextField/TextField.component';

interface ITextFieldHooked {
  isError?: boolean;
}

export const TextFieldHooked: React.FC<ITextFieldHooked> = ({
  isError,
}: ITextFieldHooked) => {
  const {
    register,
    setError,
    formState: { errors },
  } = useForm<{ email: string }>();

  React.useEffect(() => {
    if (isError)
      setError('email', {
        type: 'required',
        message: 'This is required',
      });
  }, [isError, setError]);

  return (
    <StyledComponentsProvider>
      <TextField
        error={errors}
        errorMessageTestID="molecule-form-input-error-message"
        label="Email"
        name="email"
        {...(register('email'), { required: true })}
      />
    </StyledComponentsProvider>
  );
};
