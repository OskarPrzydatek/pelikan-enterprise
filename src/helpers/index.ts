export { DateFieldHelper } from './DateField.helper';
export { SelectHelper } from './Select.helper';
export { TextAreaHelper } from './TextArea.helper';
export { TextFieldHelper } from './TextField.helper';
