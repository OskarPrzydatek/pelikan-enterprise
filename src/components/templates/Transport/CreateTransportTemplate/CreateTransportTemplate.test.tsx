import { render } from '@testing-library/react';
import { describe, test, vi } from 'vitest';

import { CreateTransportTemplateHelper } from '~/helpers';

const mockOnSubmitCreateTransport = vi.fn();

describe('CreateTransport Template', () => {
  test('component snapshot', () => {
    const view = render(
      <CreateTransportTemplateHelper
        onSubmitCreateTransport={mockOnSubmitCreateTransport}
      />
    );
    expect(view).toMatchSnapshot();
  });
});
