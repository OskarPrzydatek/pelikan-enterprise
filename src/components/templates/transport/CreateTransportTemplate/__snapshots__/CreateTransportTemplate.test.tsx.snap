// Vitest Snapshot v1

exports[`CreateTransport Template > component snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-cjibBx bSlRRw"
      >
        <form
          class="sc-ftTHYK jGKFAf"
        >
          <h2
            class="sc-dkrFOg kaKqDq"
          >
            Dodaj transport
          </h2>
          <div
            class="sc-bjfHbI kTUcsA"
          >
            <div
              class="sc-idXgbr enkruV"
            >
              <input
                class="sc-dIfARi cqofnP"
                data-testid="create-company-name"
                id="companyName"
                name="companyName"
                placeholder="Nazwa firmy transportowej"
              />
            </div>
          </div>
          <div
            class="sc-dmctIk iThEeJ"
          >
            <div
              class="sc-kgTSHT fmvgAY"
            >
              <span
                class="sc-bBABsx kBxwHb"
              >
                <svg
                  data-testid="chevron-icon"
                  fill="none"
                  viewBox="0 0 27 16"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M2 2L13.5 13L25 2"
                    stroke="#013059"
                    stroke-width="3"
                  />
                </svg>
              </span>
              <select
                class="sc-iveFHk eqISZE"
                data-testid="create-transport-transport-kind"
                name="transportType"
              >
                <option
                  class="sc-cwSeag"
                  disabled=""
                  selected=""
                  value=""
                >
                  Rodzaj transportu
                </option>
                <option
                  class="sc-cwSeag"
                  value="TRANSPORT_A"
                >
                  Transport A
                </option>
                <option
                  class="sc-cwSeag"
                  value="TRANSPORT_B"
                >
                  Transport B
                </option>
                <option
                  class="sc-cwSeag"
                  value="TRANSPORT_C"
                >
                  Transport C
                </option>
              </select>
            </div>
          </div>
          <div
            class="sc-ksBlkl gBSrBs"
          >
            <div
              class="sc-hBxehG eptxaL"
            >
              <textarea
                class="sc-fEXmlR kcJGhn"
                data-testid="create-transport-desription"
                id="description"
                name="description"
                placeholder="Opis"
              />
            </div>
          </div>
          <input
            class="sc-pyfCe hqBgck"
            type="submit"
            value="Dodaj"
          />
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-cjibBx bSlRRw"
    >
      <form
        class="sc-ftTHYK jGKFAf"
      >
        <h2
          class="sc-dkrFOg kaKqDq"
        >
          Dodaj transport
        </h2>
        <div
          class="sc-bjfHbI kTUcsA"
        >
          <div
            class="sc-idXgbr enkruV"
          >
            <input
              class="sc-dIfARi cqofnP"
              data-testid="create-company-name"
              id="companyName"
              name="companyName"
              placeholder="Nazwa firmy transportowej"
            />
          </div>
        </div>
        <div
          class="sc-dmctIk iThEeJ"
        >
          <div
            class="sc-kgTSHT fmvgAY"
          >
            <span
              class="sc-bBABsx kBxwHb"
            >
              <svg
                data-testid="chevron-icon"
                fill="none"
                viewBox="0 0 27 16"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M2 2L13.5 13L25 2"
                  stroke="#013059"
                  stroke-width="3"
                />
              </svg>
            </span>
            <select
              class="sc-iveFHk eqISZE"
              data-testid="create-transport-transport-kind"
              name="transportType"
            >
              <option
                class="sc-cwSeag"
                disabled=""
                selected=""
                value=""
              >
                Rodzaj transportu
              </option>
              <option
                class="sc-cwSeag"
                value="TRANSPORT_A"
              >
                Transport A
              </option>
              <option
                class="sc-cwSeag"
                value="TRANSPORT_B"
              >
                Transport B
              </option>
              <option
                class="sc-cwSeag"
                value="TRANSPORT_C"
              >
                Transport C
              </option>
            </select>
          </div>
        </div>
        <div
          class="sc-ksBlkl gBSrBs"
        >
          <div
            class="sc-hBxehG eptxaL"
          >
            <textarea
              class="sc-fEXmlR kcJGhn"
              data-testid="create-transport-desription"
              id="description"
              name="description"
              placeholder="Opis"
            />
          </div>
        </div>
        <input
          class="sc-pyfCe hqBgck"
          type="submit"
          value="Dodaj"
        />
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
